### ISSUE TITLE
I found a freezing bug across all xanmod-5.15 kernels

### ISSUE THREAD

**Issue by skulldealer** [xanmod/linux#212]
hi, i am the guy who reported this few days back:
https://github.com/hamadmarri/TT-CPU-Scheduler/issues/3

Alexandre Frade found out it was an elementary os problem after checking my logs. It's incorrect !
(Reference: https://github.com/hamadmarri/TT-CPU-Scheduler/issues/3#issuecomment-962674264  )

Let me explain,
the freeze that happened while i reported the issue was when i was rapidly switching workspaces on elementary os.
this might explain why the logs pointed to something related to elementary os.
i use zoom meetings on chromium based browsers, without switching workspaces. i also leave my computer idle with btop system monitor on with full screen (F11 key). after running these 2 things for 10 to 15 minutes under xanmod 5.15 edge and cacule-tt kernel, it again freezes. if logs are checked then probably it will point towards some problems other than elementay os. i am yet to test 5.15 generic unsigned kernel. But i have tested the deault elementary os kernel (version linux-image-5.11.0-40-generic/focal-updates,now 5.11.0-40.44~20.04.2 amd64) and it works perfectly fine without any system freezing. this makes it harder to believe that it is a elementary os specific problem. Also when i was using  5.13 and 5.14 xanmod cacule kernels, i had such freezes very very ocassionally but after 5.15 its atleast happening 20 times a day. so either its a xanmod patch or something with 5.15 release itself causing the problem. I will take some time to test the generic unsigned 5.15 kernel. But i felt like reporting the issue. 
I can easily replicate the freeze whenever i want. how do i report it to you? I am sure it is a kernel problem with my hardware.
i have 6 year old non-unique hardware. Lenovo Yoga 300-11IBR. I have experienced this freezing problem ocassionally on pop os also. I have ignored it many times, but this time i want to solve it. Plese help me.

I have tested many distros, also tried running many tests. I am 100% sure it is a software problem and all my hardware is healthy and fine.

Also I have noticed that the freeze happens when some real time task is going on. Like a zoom video call, actively monitoring system, scrolling web, switching workspaces, etc. If I boot my computer, log in, and keep it as it is it will never freeze.

**Comment by skulldealer** [comment]
What logs should I send you?

**Comment by skulldealer** [comment]
I am using 5.11 kernel on elementary os 6 based on ubuntu 20.04 lts.
I have not encountered any freeze on this kernel. But I encounter a freeze everytime, few minutes after booting any xanmod 5.15 kernel. I am sure it is not an elementary os 6 issue, but related to 5.15 release or some xanmod patch.

**Comment by xanmod** [comment]
Freezes with 5.15 generic?

**Comment by skulldealer** [comment]
> Freezes with 5.15 generic?

No. 5.15 generic kernel doesn't freeze,
5.11.0-40.44~20.04.2 also doesn't freeze.
5.15 edge freezes sometimes(still not reliable),
5.15-tt cacule freezes the most, everytime after booting within 5 to 10 minutes

**Comment by monarc99** [comment]
i have short freezes with 5.15.1-tt and Ananicy-Cpp. After i disabled ananicy-cpp and rebooted it seems to run fine now.

Haven't tried ananicy-cpp again. (for double check)  But i could, if requested.

**Comment by skulldealer** [comment]
> i have short freezes with 5.15.1-tt and Ananicy-Cpp. After i disabled ananicy-cpp and rebooted it seems to run fine now.
> 
> Haven't tried ananicy-cpp again. (for double check)  But i could, if requested. 

Wow. That was a good tip. Let me try this for some time before commenting on the difference it makes.
But if that's true, maintainers of Garuda linux are gonna be in trouble as they use ananicy-cpp and a many of its users use much recent cacule kernel directly compiled from aur.

**Comment by monarc99** [comment]
If you test it, its not enough only to disable ananicy-cpp, a reboot is needed. Only after that my freezes were gone.

**Comment by lavilao** [comment]
Hi did you found a solution to this issue? I have been with system hangs since linux-zen updated to 5.15.3 all the way up to 5.15.5. Just installed linux-xanmod-tt-5.15.5 to test and it also hang. Did disabling ananicy-cpp fixed? (I am currently using garuda sway so ananicy-cpp is a common variable)

**Comment by monarc99** [comment]
There a several bugs, that can cause a hang with the 5.15 kernel.

Do you have messages in journalctl log? maybe from i915 driver?

**Comment by lavilao** [comment]
this is the output of the garuda assistant journalctl error
~~~text
-- Journal begins at Sun 2021-11-21 13:03:27 CST, ends at Sat 2021-11-27 19:21:16 CST. --
nov 27 17:59:30 luis-x453sa kernel: mmc0: card never left busy state
nov 27 17:59:30 luis-x453sa kernel: mmc0: error -110 whilst initialising SD card
nov 27 17:59:30 luis-x453sa kernel: mmc0: card never left busy state
nov 27 17:59:30 luis-x453sa kernel: mmc0: error -110 whilst initialising SD card
nov 27 18:02:15 luis-x453sa kernel: ntfs3: Unknown parameter 'windows_names'

~~~

**Comment by groobybugs** [comment]
Hi guys!

Yes, since version 15.3 I have had this problem, like 3 or 4 times in my daily use. The only solution is restart the machine, basically the same case as @skulldealer , I don't know what triggers this problem, maybe it's when I'm having meetings, it's quite strange.

**Comment by lavilao** [comment]
I just installed manjaro gnome (previously on garuda sway) and it seems fine? did the problem really was ananicy-cpp?

**Comment by cj0nes** [comment]
I am running Pop_OS! 21.04 and have recently updated to 5.15.8-rt23-xanmod1. I've also been having infrequent and inexplicable freezes. I am not sure how to reproduce the freezing as it seems to happen when I'm doing different things. I have looked through kern.log, syslog, and others but have not come across anything promising to provide.

I updated from 5.11.12-rt11-xanmod1 which I ran for months without issue. Let me know how I can help.
